docs/io/fits/index.rst

```diff
@@ -309,6 +309,78 @@ would with a dict::
 
     See also :ref:`sphx_glr_generated_examples_io_modify-fits-header.py`.
 
+.. _structural_keywords:
+
+Structural Keywords
+"""""""""""""""""""
+
+FITS keywords mix up both metadata and critical information file structure that is
+needed to parse the file. These *structural* keywords can be silently modified by
+:mod:`astropy.io.fits` and, in general, should not be touched by the user.
```

Comment:
The important message is that these structural keywords should not be modified manually because the HDU/Header objects will manage the keywords. So I think that these attributes should be described together with the keywords. The table below is a good start, but it would be even better to tell which attribute is linked to a given keyword. E.g. NAXIS* is determined by `.data.shape`, BITPIX by `.data.dtype`, etc. Some keywords can just be ignored (SIMPLE, PCOUNT, GCOUNT). Table keywords are linked to the `Column` attributes (http://docs.astropy.org/en/latest/io/fits/usage/table.html#column-creation).

Reply:
I really like this idea. Once we decide where this table will go (see above) I'll add this additional information. Thanks for the link to the Table stuff.

Reply:
I think the long-term goal should be to make the structural headers unsettable in normal use (I can imagine a need for doing it deeper in the implementation, or when constructing unusual FITS file formats).  But that seems like potentially a great deal of development effort, perhaps disproportionate to its value, so I do support as an interim measure (potentially one that might be around for years - let's be realistic) having "what not to do" documentation, perhaps of the form "In general, one should not attempt to set the headers in Table X directly, as they must remain consistent with the properties of the actual data payload of the file."  I think having the table mentioned above is an excellent idea in this context.